include/swift/AST/NameLookup.h

```diff
@@ -809,6 +809,10 @@ class ASTScope : public ASTAllocated<ASTScope> {
   static std::pair<CaseStmt *, CaseStmt *>
   lookupFallthroughSourceAndDest(SourceFile *sourceFile, SourceLoc loc);
 
+  /// Returns \c true if the given source location is inside an attached
+  /// or freestanding macro argument.
+  static bool isInMacroArgument(SourceFile *sourceFile, SourceLoc loc);
```

Comment:
Per my previous comment, since we need some level of name lookup to evaluate this predicate, can we move it alongside the name lookup operations? It makes the layering more clear.